x-pack/plugins/reporting/server/usage/index.d.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface AvailableTotal {
+  available: boolean;
+  total: number;
+}
+
+interface StatusCounts {
+  [statusType: string]: number;
+}
+
+export interface KeyCountBucket {
+  key: string;
+  doc_count: number;
+}
+
+export interface AggregationBuckets {
+  buckets: KeyCountBucket[];
+  pdf?: {
+    buckets: KeyCountBucket[];
+  };
+}
+
+/*
+ * Mapped Types and Intersection Types
+ */
+
+type AggregationKeys = 'jobTypes' | 'layoutTypes' | 'objectTypes' | 'statusTypes';
+export type AggregationResults = { [K in AggregationKeys]: AggregationBuckets } & {
+  doc_count: number;
+};
+
+type RangeAggregationKeys = 'all' | 'lastDay' | 'last7Days';
+export type RangeAggregationResults = { [K in RangeAggregationKeys]?: AggregationResults };
+
+type BaseJobTypeKeys = 'csv' | 'PNG';
+export type JobTypes = { [K in BaseJobTypeKeys]: AvailableTotal } & {
+  printable_pdf: AvailableTotal & {
+    app: {
+      visualization: number;
+      dashboard: number;
+    };
+    layout: {
+      print: number;
+      preserve_layout: number;
+    };
+  };
+};
+
+export type RangeStats = JobTypes & {
+  _all: number;
+  status: StatusCounts;
+};
+
+export type UsageObject = RangeStats & {
+  available: boolean;
+  enabled: boolean;
+  browser_type: string;
+  lastDay: RangeStats;
+  last7Days: RangeStats;
+};
+
+export type ExportType = 'csv' | 'printable_pdf' | 'PNG';
```

Comment:
is PNG supposed to be lower case?  I see some conversion code between pretty print and id, and the ids are lower case?

Reply:
PNG unfortunately is lowercase. The feature was implemented with the export type ID in uppercase, and it's the only one that has it that way.